qa327/frontend.py

```diff
@@ -28,7 +29,7 @@ def register_post():
     if password != password2:
         error_message = "The passwords do not match"
 
-    elif len(email) < 1:
+    elif (len(email) < 1 or not re.match(r"^[A-Za-z0-9\.\+_-]+@[A-Za-z0-9\._-]+\.[a-zA-Z]*$", email)):
```

Comment:
This is a good line of code to validate the email format. 

However, I'm seeing this twice in this file. I think this line could be modularized into a function that returns `True` if the email is valid or `False` if the email is not valid. This ensures we follow DRY principles and makes the code cleaner too.

Then you could call it here like:
```
if not function:
    # error message and/or re-render login page 
 ```
which you can do again the second time you need to check email format.